### DISCUSSION TITLE
Mechanism to Phase Out Deprecated/Broken/Abandoned Crates

### DISCUSSION THREAD

**Discussion by justkash** [rust-lang/crates.io#5064] [Feature Requests]
In an effort to ensure that the Rust Crate ecosystem doesn't get bloated with deprecated crates, we need to establish a mechanism for phasing out crates that are in broken, deprecated, or abandoned states. These crates not only take up namespaces but also increase the search space for Rust users when searching for libraries and tools. This problem is especially exasperated for crates related to cryptography where the use of deprecated or vulnerable crates may have drastic consequences.

I am unsure what the final user experience will look like, but perhaps crates can be placed into one of a few stages that denote their activity level or acceptance into the Rust ecosystem.

**Comment by carols10cents** [comment]
This needs an RFC, not a discussion here. If you're looking to discuss ideas before opening an RFC, [the internals forum](https://internals.rust-lang.org/) is the place for that. Thanks!
